site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -132,65 +143,79 @@ public function submitGrade() {
         $who_id = $_POST['u_id'];
         $gradeable = $this->core->getQueries()->getGradeable($gradeable_id, $who_id);
 
+        if ($this->core->getUser()->getGroup() > $gradeable->getMinimumGradingGroup()) {
+            $_SESSION['messages']['error'][] = "You do not have permission to grade {$gradeable->getName()}";
+            $this->core->redirect($this->core->getConfig()->getSiteUrl());
+        }
+
         $now = new \DateTime('now', $this->core->getConfig()->getTimezone());
         $homeworkDate = $gradeable->getGradeStartDate();
         if ($now < $homeworkDate) {
             $_SESSION['messages']['error'][] = "Grading is not open yet for {$gradeable->getName()}";
             $this->core->redirect($this->core->buildUrl(array('component'=>'grading', 'page'=>'electronic', 'gradeable_id' => $gradeable_id)));
         }
+
+        if (isset($_POST['overwrite'])) $gradeable->setGrader($this->core->getUser());
```

Comment:
HW reports and the overwrite grader checkbox on the TA grading page still use the grader stored in the gradeable model so this has to stay for now. I think Marisa is currently working on that.